kotlintest-tests/kotlintest-tests-core/src/test/kotlin/com/sksamuel/kotlintest/matchers/MatchersTest.kt

```diff
@@ -15,6 +19,33 @@ import kotlin.collections.HashMap
 
 class MatchersTest : FreeSpec({
 
+  "withClue()" - {
+
+    "should prepend clue to message" {
+      try{
+        withClue("a clue:") { listOf(1) shouldHaveSize 2 }
```

Comment:
I'm a little bit afraid of "hardcoding" the exception this way. What if the `shouldHaveSize` message changes? We'll have another point of code to take care of, while `withClue` will keep working.

Technically, this is testing two things: That Clue works and that Collection `shouldHaveSize` message is x

Reply:
Fixed. I created a custom matcher just for the test